src/components/DateRangeInput/PDateRangeInput.vue

```diff
@@ -1,133 +1,93 @@
-      <p-button
-        class="p-date-range-input__date"
-        :class="classes.date(date, today, outOfMonth)"
-        :disabled="disabled"
-        small
-        flat
-        @click="select"
-      >
-        <slot
-          name="date"
-          :date="date"
-          :is-selected="isDateSelected(date)"
-          :is-in-selected-range="isDateInSelectedRange(date)"
+
+    <template #date="{ date, disabled: dateDisabled, today, inMonth }">
+      <div class="p-date-range-input__date-wrapper" :class="classes.dateWrapper(date)">
+        <p-button
+          class="p-date-range-input__date"
+          :class="classes.date(today, inMonth)"
+          :disabled="dateDisabled"
+          small
+          flat
+          @click="setDate(date)"
         >
-          {{ date.getDate() }}
-        </slot>
-      </p-button>
+          <slot
+            name="date"
+            :date="date"
+            :is-selected="isSelectedDate(date)"
+            :is-in-selected-range="isDateInSelectedRange(date)"
+          >
+            {{ date.getDate() }}
+          </slot>
+        </p-button>
+      </div>
+    </template>
+
+    <template #controls>
+      <PContent>
+        <PDateTimeInputGroup v-model="internalStartDate" :show-time="showTime" label="Start Date" />
+        <PDateTimeInputGroup v-model="internalEndDate" :show-time="showTime" label="End Date" />
+      </PContent>
     </template>
   </PDateInput>
 </template>
 
 <script lang="ts" setup>
   import { format, isSameDay, startOfDay, endOfDay } from 'date-fns'
-  import { computed } from 'vue'
+  import { computed, ref } from 'vue'
+  import PContent from '@/components/Content/PContent.vue'
   import PDateButton from '@/components/DateInput/PDateButton.vue'
   import PDateInput from '@/components/DateInput/PDateInput.vue'
+  import PDateTimeInputGroup from '@/components/DateTimeInputGroup/PDateTimeInputGroup.vue'
   import { isDateBefore, isDateInRange } from '@/utilities'
-  import { media } from '@/utilities/media'
 
   const props = defineProps<{
     startDate: Date | null | undefined,
     endDate: Date | null | undefined,
     viewingDate?: Date,
+    showTime?: boolean,
     clearable?: boolean,
+    disabled?: boolean,
+    min?: Date | null | undefined,
+    max?: Date | null | undefined,
   }>()
 
   const emit = defineEmits<{
-    (event: 'update:startDate' | 'update:endDate', value: Date | null): void,
+    (event: 'update:startDate' | 'update:endDate', value: Date | null | undefined): void,
     (event: 'update:viewingDate', value: Date | undefined): void,
   }>()
 
-  const dateFormat = 'MMM do, yyyy'
-
-  const internalStartDate = computed({
-    get() {
-      return props.startDate ?? null
-    },
-    set(value) {
-      emit('update:startDate', value)
-    },
-  })
+  // this dummy ref is bound to PDatInput's v-model but isn't directly used for anything
+  // this component tracks its own dates
+  const dummy = ref()
```

Comment:
This looked like a dev relic till I saw this comment - maybe rename this var to `internalModelValue` to avoid confusion? then the comment becomes obsolete potentially

Reply:
I'm not apposed to that. Its just a little different. An internalModelValue I would expect to match the modelValue prop which there is none in this component (there's a startDate and endDate v-model). Also this literally isn't used for anything. This component literally overrides all the functionality of the PDateInput via its slots. But PDateInput requires a v-model. So this is only here to satisfy that constraint. IMO `internalModelValue` doesn't convey that. But will update if you feel strongly.